conda/gateways/repodata/jlap/fetch.py

```diff
@@ -0,0 +1,402 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+# Lappin' up the jlap
+from __future__ import annotations
+import io
+
+import json
+import logging
+import pathlib
+import pprint
+import re
+import time
+import zstandard
+from contextlib import contextmanager
+from hashlib import blake2b
+from typing import Iterator
+
+import jsonpatch
+from requests import HTTPError
+
+from conda.base.context import context
+from conda.gateways.connection import Response, Session
+from conda.gateways.repodata import RepodataState
+
+from .core import JLAP
+
+log = logging.getLogger(__name__)
+
+
+DIGEST_SIZE = 32  # 160 bits a minimum 'for security' length?
+
+JLAP_KEY = "jlap"
+HEADERS = "headers"
+NOMINAL_HASH = "nominal_hash"
+ON_DISK_HASH = "actual_hash"
+LATEST = "latest"
+JLAP_UNAVAILABLE = "jlap_unavailable"
+ZSTD_UNAVAILABLE = "zstd_unavailable"
+
+
+def hash():
+    """
+    Ordinary hash.
+    """
+    return blake2b(digest_size=DIGEST_SIZE)
```

Comment:
This is a constant. Why isn't it stored as one?

Reply:
This creates a new hasher object each time.

Reply:
Maybe I'm missunderstanding but I see this:

```python
>>> from hashlib import blakeb2
>>> blake2b(digest_size=32).digest() == blake2b(digest_size=32).digest()
True
```

Reply:
something like this

```
b2 = blake2b(digest_size=32)
b2.update(b'some data')
b2.digest()

b2.update(b'some data')
b2.digest()
```

Reply:
ah, of course

what do we gain by defining this `hash` helper function instead of inlining the `blake2d` initialization?

Reply:
we never pass the wrong digest size by mistake, and could change the hash function to blake3 or whatever if it becomes availabel